sql-odbc/src/sqlodbc/opensearch_communication.cpp

```diff
@@ -900,13 +904,16 @@ std::string OpenSearchCommunication::GetServerDistribution() {
 
     std::shared_ptr< Aws::Http::HttpResponse > response =
         IssueRequest("", Aws::Http::HttpMethod::HTTP_GET, "", "", "");
-    if (response == nullptr) {
+    if (response == nullptr || response->GetResponseCode() == Aws::Http::HttpResponseCode::REQUEST_NOT_MADE) {
         m_error_message =
             "Failed to receive response from server version query. "
-            "Received NULL response.";
+            "Received no response from url: "
+            + (m_rt_opts.conn.server + (m_rt_opts.conn.port.empty() ?
+            "" : ":" + m_rt_opts.conn.port));
         SetErrorDetails("Connection error", m_error_message,
                         ConnErrorType::CONN_ERROR_COMM_LINK_FAILURE);
         LogMsg(OPENSEARCH_ERROR, m_error_message.c_str());
+        m_error_message_to_user = m_error_message;
         return "";
```

Comment:
why are we returning "" in all cases?  Does it make sense to reconsider the response type of this function?  Change to return void, or change the returns so that they return actual content ....?

Reply:
Note: this is the problem with not have a javadoc. 
The API contract for this function isn't described, which means I have to read the code and look up all the usages of this function to understand WHAT string is expected from this function. 
Please add a javadoc header comment to the function you know what the parameters/returns are supposed to be. Then determine if a string return type is appropriate.